Vagrantfile

```diff
@@ -0,0 +1,22 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure(2) do |config|
+  config.vm.box = "ubuntu/trusty64"
+  config.vm.network "forwarded_port", guest: 8123, host: 8123
+  config.vm.synced_folder ".", "/vagrant"
+  config.vm.provision "shell", inline: <<-SHELL
```

Comment:
Thats going beyond the initial scope of this pull request. In my experience vagrant has been a tool to get up and running. It doesn't have the concept of running a conditional command. 

In ruby projects I work on, vagrant is the provisioning tool, and the running of tests and restarting services is typically handled by a [Rakefile](https://github.com/ruby/rake) and a [Guardfile](https://github.com/guard/guard/wiki/Guardfile-examples)

I haven't done much with python, but it sounds like it has similar concepts like paver, and livereload. 

- [paver](https://github.com/paver/paver)
- [live reload](https://github.com/lepture/python-livereload)

I think that would be best handled in a separate merge request from someone with more python experience than me.